src/upnp.cpp

```diff
@@ -708,6 +708,22 @@ void upnp::post(upnp::rootdevice const& d, char const* soap
 #endif
 }
 
+std::string upnp::create_soap(char const* soap_action, std::string const& service_namespace, char const* part)
```

Comment:
maybe you could take a ``std::initialization_list`` of (tag-name, value) pairs and format the whole message in here